### DISCUSSION TITLE
Uno Binding programatically in C#

### DISCUSSION THREAD

**Discussion by Ray6402** [unoplatform/uno#12532] [Q&A]
Hi,
Does anyone know how to do binding for Uno programmatically in C#?

![image](https://github.com/unoplatform/uno/assets/113428244/a890a1f3-cc7c-48ce-a160-073ad6ce9f82)

**Comment by FrozDark** [comment]
Use Binding class
https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.data.binding?view=winrt-22621

**Reply by JohnTasler** [reply]
But, do note that you want to use the one from Uno.UI.dll, which is in the **_Microsoft_**.UI.Xaml.Data namespace.

**Reply by jeromelaban** [reply]
@JohnTasler `BindingBase` is available in Uno. It may be an intellisense issue. If you can provide a repro, it will help troubleshoot further.

**Reply by JohnTasler** [reply]
@jeromelaban Yes, I realize that. I was just pointing out that the namespace is **Microsoft**... and not **WIndows...**

Therefore, the link would be:
[Microsoft.UI.Xaml.Data.Binding](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.data.binding?view=windows-app-sdk-1.3)

**Reply by jeromelaban** [reply]
Indeed it should! Sometimes intellisense is lost with the resolution of some types, which may be related to nuget package restore, or late resolution not happening. In most cases, restarting VS after all package have been restore is enough to get this sorted out.

@Ray6402 if you still get these errors, could you try building the solution and [provide us with a binlog](https://platform.uno/docs/articles/uno-builds-troubleshooting.html) so we can troubleshoot further?

**Comment by baskren** [comment]
@FrozDark : This is something I do a lot, as I prefer using C# markup instead of XAML.  I've wrote [a library for this purpose](https://github.com/baskren/P42.Uno.Markup) but I've yet to document it.  That being said, the extension method I use to make binding via code can be found here:  https://github.com/baskren/P42.Uno.Markup/blob/main/P42.Uno.Markup/DependencyObjectExtensions.cs

Basic usage is something like:

``` csharp
    var textBox = new TextBlock()
                                .Bind(TextBlock.TextProperty, DataContext, nameof(MyModelClass.Name));
```

It supports converters, two-way binding, etc.  If you're curious, I would be happy to share more.
